x-pack/agent/pkg/agent/application/managed_mode.go

```diff
@@ -113,7 +113,17 @@ func newManaged(
 		return nil, errors.New(err, "fail to initialize pipeline router")
 	}
 
-	emit := emitter(log, router)
+	injectAPIKey, err := injectESOutputAPIKey(cfg.API.AccessAPIKey)
+	if err != nil {
+		return nil, err
+	}
+
+	emit := emitter(
+		log,
+		router,
+		injectAPIKey,
+		injectMonitoring,
```

Comment:
hah

Reply:
The way you have structured this, we could also have a dynamicDecorator where a spec can define which decorator need to be applied..